lib/matplotlib/image.py

```diff
@@ -168,11 +169,28 @@ def _resample(
     allocating the output array and fetching the relevant properties from the
     Image object *image_obj*.
     """
+
+    # decide if we need to apply anti-aliasing if the data is upsampled:
+    # compare the number of displayed pixels to the number of
+    # the data pixels.
+    interpolation = image_obj.get_interpolation()
+    if interpolation == 'antialiased':
+        # don't antialias if upsampling by an integer number or
+        # if zooming in more than a factor of 3
+        # do antialiasing....
+        # compare the number of displayed pixels of the image to the number of
+        # the data pixels.
+        disppixels = int(transform.transform([data.shape[1], 0])[0])
+        if (disppixels < 3 * data.shape[1] and disppixels != data.shape[1]):
```

Comment:
compare in both x and y and antialias if at least one of them is below nyquist?

Reply:
The filters are isotropic, so I don't know what we should do if the up/down sampling isn't isotropic.

Reply:
then at least the condition should either be "if both are <3x" or "if one is <3x", not "if the x-axis is <3x."

Reply:
But thats what I mean - the x re-sample rate should be the same as the y, unless someone is using a funky transform?

Reply:
like, its just extra code, so happy to include it if you like...

Reply:
One could just be doing `imshow([[0, 1], [2, 3]], aspect="auto")` for example.

Reply:
OK, did the y case as well, and added the check for 2x upsample